subprojects/docs/src/docs/userguide/authoring-builds/multi-project/sharing_build_logic_between_subprojects.adoc

```diff
@@ -0,0 +1,72 @@
+= Sharing Build Logic between Subprojects
+
+[[sec:convention_plugins]]
+== Convention Plugins
+
+Usually, subprojects in a multi-project build share some common traits.
+For example, several subprojects may contain _code_ in a particular programming language while another subproject may be
+dedicated for _documentation_.
+Code quality rules apply to all of the _code_ subprojects but not the _documentation_ subproject.
+At the same time, the subprojects that share one common trait may serve different purposes - they may produce different _artifact types_ that further differentiate them, for example:
+
+- _public libraries_ - libraries that are published to some repository
+- _internal libraries_ - libraries on which other subprojects depend on internally within the project
+- _command line applications_ - applications with specific packaging requirements
+- _web services_ - applications with specific packaging requirements that are different from above
+- etc
+
+Some other _code_ subprojects may be dedicated for _testing purposes_ and so on.
+
+The traits above identify a subproject's _type_.
+Or in other words, a subproject's _type_ tells us what traits the project has.
+
+Gradle's recommended way of organizing build logic is to use its plugin system.
+A plugin should define the _type_ of a subproject.
+In fact, <<plugin_reference#,Gradle core plugins>> are modeled in the same way - for example, the <<java_plugin#,Java Plugin>> configures a generic `java` project,
+while <<java_library_plugin#,Java Library Plugin>> internally applies the <<java_plugin#,Java Plugin>> and configures aspects specific to a Java library in addition.
+Similarly, the <<application_plugin#,Application Plugin>> applies and configures the <<java_plugin#,Java Plugin>> and the <<distribution_plugin#,Distribution Plugin>>.
+
+You can compose custom build logic by applying and configuring both core and external plugins and create custom plugins
+that define new project _types_ and configure conventions specific to your project or organization.
+For each of the example traits from the beginning of this section, we can write a plugin that encapsulates
+the logic common to the subproject of a given _type_.
+
+We recommend putting source code and tests for the convention plugins in the special `buildSrc` directory in the root directory of the project.
+For more information about `buildSrc`, consult <<organizing_gradle_projects.adoc#sec:build_sources,Using buildSrc to organize build logic>>.
+
+Have a look at the link:../samples/sample_convention_plugins.html[sample that demonstrates a multi-project build that models the build logic using convention plugins].
+
+Another, more complex and real-world example of a multi-project build that composes build logic using convention plugins
+is the build of the link:https://github.com/gradle/gradle[Gradle Build Tool] itself.
+
+[[sec:convention_plugins_vs_cross_configuration]]
+== Convention plugins versus cross project configuration
```

Comment:
```suggestion
== Cross project configuration
```

With the other section gone, I think we can call this just "cross project configuration"